src/OpenFga.Sdk/ApiClient/BaseClient.cs

```diff
@@ -124,9 +121,10 @@ public async Task<ResponseWrapper<T>> SendRequestAsync<T>(HttpRequestMessage req
 
             T responseContent = default;
             if (response.Content != null && response.StatusCode != HttpStatusCode.NoContent) {
-                responseContent = await response.Content.ReadFromJsonAsync<T>(cancellationToken: cancellationToken)
-                                      .ConfigureAwait(false) ??
-                                  throw new FgaError();
+                var contentString = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                if (!string.IsNullOrEmpty(contentString)) {
+                    responseContent = JsonSerializer.Deserialize<T>(contentString) ?? throw new FgaError();
+                }
```

Comment:
Pass the response stream into DeserializeAsync rather than doing this to avoid an intermediary allocation.